.github/workflows/sycl-linux-run-tests.yml

```diff
@@ -218,6 +218,15 @@ jobs:
         ref: ${{ inputs.devops_ref || inputs.ref }}
         sparse-checkout: |
           devops
+
+    # If the devops dir above was checked out from devops_ref - it may contain
+    # inappropriate devops/cts_exclude_filter/. Check out from testing branch.
+    - name: Checkout cts_exclude_filter from testing branch
+      if: inputs.tests_selector == 'cts' && inputs.devops_ref != ''
+      run: |
+        git fetch origin ${{ inputs.ref }}
+        git checkout origin/${{ inputs.ref }} -- devops/cts_exclude_filter
```

Comment:
That looks very hacky. First, I think we should move that logic to `devops/actions/run-tests/cts/`. Second, maybe put filter outside `devops/`?

Reply:
> we should move that logic to devops/actions/run-tests/cts/

I don't mind, but then we need one more input for `cts/action.yml` like `filter_ref` what I wanted to avoid. Should we proceed with it?

> maybe put filter outside devops/?

No objections, but `actions/checkout(sparse-checkout: cts_exclude_filter)` will anyway delete devops dir if this is what you mean.

Reply:
What I meant is that if that file should be used from a revision different than that used for devops/, then it does not belong to devops.

Reply:
> I don't mind, but then we need one more input for `cts/action.yml` like `filter_ref` what I wanted to avoid. Should we proceed with it?

Why doesn't it match some other "ref" that we already pass?

Reply:
> > I don't mind, but then we need one more input for `cts/action.yml` like `filter_ref` what I wanted to avoid. Should we proceed with it?
> 
> Why doesn't it match some other "ref" that we already pass?

I mean this is a ref to checkout the SYCL-CTS repo:
https://github.com/intel/llvm/blob/7bdb7580fb66d37b3a13629274667e876a547860/devops/actions/run-tests/cts/action.yml#L4-L7
If we want to move this code here, we need one more ref input in this file. Is it okay?

Reply:
IMO, yes. We can keep them as `devops_ref`, `compiler_ref`, `tests_ref`/`ref`.

Reply:
@aelovikov-intel please verify when you return to work

Reply:
> Second, maybe put filter outside `devops/`?

I'm against moving SYCL-CTS filters to a root directory. Probably, they should be moved to `sycl/` directory tree.

Reply:
@bader can we merge this one and then proceed with the reorganization or do you see it somehow differently?

Reply:
@KornevNikita, https://github.com/intel/llvm/pull/17228#issuecomment-2704380165 answers your question.

Reply:
> @aelovikov-intel please verify when you return to work

looks reasonable.

> We have at three different test directories inside sycl/ directory. This change adds another directory related to SYCL project testing.
I suggest we move all tests to sycl/test sub-directory. Existing sycl/test/* tests can be moved to a separate sub-directory alongside with e2e, cts_config and unittests.

I think `/test` and `/unittests` mirrors upstream setup, so I'd argue we should keep the same. For `test` vs `test-e2e`, I've been thinking about it a bit. With the recent build/run-only modes for e2e we would be able to say that "test" is now a sub-part of e2e, but I wasn't sure if there is much of ROI for the unification.